src/http-adapter.js

```diff
@@ -76,19 +77,32 @@ class HttpAdapter {
    * @returns {Promise} Promise for destroy.
    */
   destroy (link) {
-    return this._ajax('DELETE', link)
+    return this._http(METHOD.DELETE, link)
   }
 
-  _ajax (type, url, data) {
-    let headers = {
+  async _http (method = METHOD.GET, url = '', data = {}) {
+    if (!url) throw new Error('url is not defined!')
+
+    const headers = {
       'Accept': 'application/vnd.api+json',
       'Content-Type': 'application/vnd.api+json'
     }
 
-    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
-    // some instrumented Vue data)
-    if (data && ['PATCH', 'POST'].indexOf(type) > -1) {
-      data = JSON.stringify(data)
+    // explicitly set body to undefined. Contrary to MDN documentation, GET methods cannot take a body. Therefore, the default must be undefined
+    let body
+
+    // eslint-disable-next-line no-undef
+    url = new URL(url, location.origin)
```

Comment:
Not a fan of this `url` string becoming an object. It looks like it switches back to a string in `_makeRequest`. 
Maybe something like:

```suggestion
    const { searchParams } = new URL(url, location.origin)
```

Reply:
I see what you mean. Fetch is really just calling the toString method on the url, putting GET methods with the url and the query params, and post methods with just the URL. We also construct the url just to get the origin of the request set. What we could do is make the object, and if its a GET request, add the query params, then call to string on it and pass it to get? That way _makeRequest receives a string

Reply:
I missed the part about adding `location.origin`. Just looking for a way to differentiate the two types. Would something like this work?

```
const url = new URL(requestUrl, location.origin)
```

Reply:
yeah we could do that. then maybe we could do after the if/else on the method block

requestUrl = url.toString()

which could allow _makeRequest to take a String an would look like `http://mydomain.com/api/thing/2/?cat=meow&dog=woof' or something like `http://mydomain.com/api/thing/2/`

Hopefully I put those together right haha

Reply:
I'm not sure leaving URL relative preserves the expected behavior. `fetch` handles relative URIs as relative to the current full location URL. So e.g. 'api/session/' called on page `https://www.grove.co/catalog/product/walnut-scrubber-sponge/` would resolve to `https://www.grove.co/catalog/product/walnut-scrubber-sponge/api/session/`, but with the original code in the PR would resolve correctly to `https://www.grove.co/api/session/`.

Reply:
I'm not sure I follow. Which piece of code is the culprit and which piece is correct?